rmf_traffic_ros2/test/unit/test_ParticipantRegistry.cpp

```diff
@@ -0,0 +1,338 @@
+
+  rmf_traffic::schedule::ParticipantDescription p1(
+    "participant 1",
+    "test_Participant",
+    rmf_traffic::schedule::ParticipantDescription::Rx::Responsive,
+    rmf_traffic::Profile{shape});
+
+  rmf_traffic::schedule::ParticipantDescription p2(
+    "participant 2",
+    "test_Participant",
+    rmf_traffic::schedule::ParticipantDescription::Rx::Responsive,
+    rmf_traffic::Profile{shape});
+  
+  rmf_traffic::schedule::ParticipantDescription p3(
+    "participant 3",
+    "test_Participant",
+    rmf_traffic::schedule::ParticipantDescription::Rx::Responsive,
+    rmf_traffic::Profile{shape});
+  
+  ParticipantId participant_id1, participant_id2, participant_id3;
+  
+  GIVEN("A stubbed out logger")
+  {
+    auto logger = std::make_shared<TestOperationLogger>();
+    WHEN("Creating a new DB without errors")
+    {
+      auto db1 = std::make_shared<Database>();
+      ParticipantRegistry registry1(logger, db1);
+      participant_id1 = registry1.add_participant(p1);
+      participant_id2 = registry1.add_participant(p2);
+      registry1.remove_participant(participant_id2);
+      participant_id3 = registry1.add_participant(p3);
+  
+      THEN("Restoring DB")
+      {
+        auto db2 = std::make_shared<Database>();
+        ParticipantRegistry registry2(logger, db2);
+        auto restored_participants = db2->participant_ids();
+        REQUIRE(restored_participants.count(participant_id1) > 0);
+        REQUIRE(restored_participants.count(participant_id2) == 0);
+        REQUIRE(restored_participants.count(participant_id3) > 0);
+        REQUIRE(restored_participants.size() == 2);
+  
+        auto _p1 = db2->get_participant(participant_id1);
+        auto _p3 = db2->get_participant(participant_id3);
+  
+        REQUIRE(*_p1 == p1);
+        REQUIRE(*_p3 == p3);
+      }
+    }
+  
+    WHEN("Creating a new DB with erroneous remove request")
+    {
+      auto db1 = std::make_shared<Database>();
+      ParticipantRegistry registry1(logger, db1);
+      participant_id1 = registry1.add_participant(p1);
+      REQUIRE_THROWS(registry1.remove_participant(1000));
+      participant_id2 = registry1.add_participant(p2);
+      registry1.remove_participant(participant_id2);
+      participant_id3 = registry1.add_participant(p3);
+  
+      THEN("Restoring DB")
+      {
+        auto db2 = std::make_shared<Database>();
+        ParticipantRegistry registry2(logger, db2);
+        auto restored_participants = db2->participant_ids();
+        REQUIRE(restored_participants.count(participant_id1) > 0);
+        REQUIRE(restored_participants.count(participant_id2) == 0);
+        REQUIRE(restored_participants.count(participant_id3) > 0);
+        REQUIRE(restored_participants.size() == 2);
+  
+        auto _p1 = db2->get_participant(participant_id1);
+        auto _p3 = db2->get_participant(participant_id3);
+  
+        REQUIRE(*_p1 == p1);
+        REQUIRE(*_p3 == p3);
+      }
+    }
+  }
+}
+
+bool file_exists(const char *fileName)
```

Comment:
We're using C++17 so you can start using `std::filesystem::exists(~)` instead of doing this.